src/content/docs/codestream/start-here/codestream-new-relic.mdx

```diff
@@ -0,0 +1,224 @@
+is produced.  The API take the following parameters:
+* `accountId` - the integer account id for the account you want to add the Repository to
+* `url` - example `https://github.com/newrelic/beta-docs-site.git`
+* `name` - example: `newrelic/beta-docs-site`
+```
+mutation {
+  referenceEntityCreateOrUpdateRepository(repositories: [{accountId: [YOUR_ACCOUNT_ID], name: "[REPO_NAME]", url: "[REPO_URL]"}]) {
+    created
+    failures {
+      guid
+      message
+      type
+    }
+  }
+}
+```
+In order to find back the entity you have created you can use a query like the following.  Note that the URL you
+provided to `referenceEntityCreateOrUpdateRepository` gets saved as an entity tag.
+```
+{
+  actor {
+    entitySearch(query: "name = 'a name' OR tags.url = 'a url'") {
+      count
+      query
+      results {
+        entities {
+          guid
+          name
+          tags {
+            key
+            values
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+**Step 2: Associate the repository entity to your application entity**
+
+First find the GUID for the application you want to associate your repository too.  
+
+Parameters:
+* `sourceEntityGuid` - the entity GUID of the application
+* `targetEntityGuid` - the entity GUID of your Repository
+* `type` - always `BUILT_FROM`
+
+```
+mutation {
+  entityRelationshipUserDefinedCreateOrReplace(sourceEntityGuid: "", targetEntityGuid: "", type: BUILT_FROM) {
+    errors {
+      message
+      type
+    }
+  }
+}
+```
+
+To see all entities related to your Repository you can do a query like this:
+```
+{
+  actor {
+    entity(guid: "[YOUR_REPOSITORY_GUID]]") {
+      relatedEntities(filter: {direction: BOTH, relationshipTypes: {include: BUILT_FROM}}) {
+        results {
+          target {
+            entity {
+              name
+              guid
+              type
+            }
+          }
+          type
+        }
+      }
+      name
+      type
+      tags {
+        values
+        key
+      }
+    }
+  }
+}
+
+```
+
+**Step 3: Cleanup (if needed)**
+
+Delete a repository with the following query:
+```
+mutation DeleteRepository {
+  entityDelete(guids: "[ENTITY_GUID_HERE]]") {
+    deletedEntities
+    failures {
+      message
+      guid
+    }
+  }
+}
+
+```
+
+  </Collapser>
+
+</CollapserGroup>
+
+## Install APM agents with CodeStream [#install-agents]
+
+Requirements for installing New Relic APM agents via CodeStream:
+* [New Relic account](https://newrelic.com/signup)
+* [New Relic user key](https://one.newrelic.com/launcher/api-keys-ui.api-keys-launcher)
+* [CodeStream and New Relic connection](#connect)
+* A supported language application codebase
+    * .NET
+    * Java
+    * Node.JS
+
+With CodeStream connected to New Relic, when you're working on an application's codebase that's not being monitored by New Relic, CodeStream will offer to instrument that application for you.
```

Comment:
A bit of a fluid situation still, but as of now, the only time CodeStream will ask the user if they want to instrument is immediately after they connect to New Relic (except when they connect as part of the Open in IDE flow). If, at any other point in the future, the user would like to instrument via CodeStream, they would go to the username menu and select Help > New Relic Setup.

Reply:
Good to know. I didn't realize this check only happened at the initial connection point. Thanks!

Reply:
Small update @paperclypse. We relocated the menu item. It's now directly under the username menu, instead of under Help.

Reply:
Got it. Thanks!